public/js/webcam.js

```diff
@@ -71,14 +79,32 @@ function postTracks(dataURL) {
             location.href = "/tracks";
         })
         .catch((error) => {
-            alert(`Error ${error.response.status}: ${error.response.statusText}`);
+            hide([loadingDiv]);
+            show([webcam, errorAlert]);
+            $("#header").hide();
+            $('#errorAlert').append('<span>'+'Sorry, we encountered an internal server error. ' + '</span>');
+
+            if(error.response) {
+                $('#errorAlert').append('<span>' + error.response.data.errorMsg +'</span>');
+
+                if(error.response.data.redirectLink) {
+                    setTimeout(function() {
+                        location.href= error.response.data.redirectLink
+                    }, 5000);
+                }
+            } else if (error.message) {
+                $('#errorAlert').append('<span>'+'Sorry, we encountered an internal server error. ' +
+                    error.message +'</span>');
+            }
+
         });
 }
 
 getTracksButton.addEventListener("click", () => {
     const dataURL = canvas.toDataURL('image/png', 1);
     hide([webcam]);
     show([loadingDiv]);
+    $("#errorAlert").html("").hide();
```

Comment:
If the first time an error happens and we append the html, what would happen the second time if an error happens again?

Would the html be reset to "" and then we append the error message, or would the error messages be appended on each other?

Reply:
`html` completely replaces the text inside the alert but `append` will add to the text! So here I'm resetting it to an empty string so the error messages don't accumulate/repeat, and hence the html resets, then appends!

Reply:
Ahh I see got it thanks! I realize we shouldn't use addEventListener, because for the same reason here! https://github.com/ericrkuo/Flow/pull/27